### DISCUSSION TITLE
parallel concordancer中索引使用KWIC来展现

### DISCUSSION THREAD

**Discussion by Sam-zcl** [BLKSerene/Wordless#25] [Q&A] [answered]
是否可以在parallel concordancer中添加以KWIC形式展现search item的选项。这样更有利于比较平行句对在关键词周围的context。

**Comment by BLKSerene** [comment]
你指的是类似于 Concordancer 的散点图吗 Parallel Concordancer 因为是平行索引
各文件的关键词位置都是一样的 直接用 Concordancer 的 Generate figure 即可

**Comment by Sam-zcl** [comment]
谢谢回复，我指的是 类似于Concodancer 中展现索引结果一样的 keywords in context的模式，也即所搜索的词汇出现在中间，左右各有n词的窗口展示。若平行语料只有句级别的对齐，可以在前述的KWIC模式后再增加一列，作为展示索引词在其他语言中的平行句对。
目前parallel concodancer展现索引结果只能以单句为单位展现，可能将索引结果以KWIC模式展现会更好比较。

**Comment by BLKSerene** [comment, marked as answer]
Parallel Concordancer 并没有自动对齐的功能 只是默认输入的平行文件是用户已经对齐的 就是一行对应一行（文本文件里的硬换行）
每行均为一个对齐单位 至于对齐单位是句子还是段落还是什么并不关心 所以显示出来的也就是用户已平行对齐的不同文件里的同一行
如果按前后n个词这样去显示 显示出来的context就不是完全对齐的了

另外平行检索的时候 比如检索源文中的某个词 是并无法得知译文里具体对应的是哪个词（node）的
也就无法计算译文中 要从哪里开始显示比如前n和后n个词 所以只能把对应的完整一行显示出来

Parallel Concordancer 在检索的时候其实也不区分源文和译文 就是在每个文件里都把所有检索项搜一遍
只要任何一个文件中检索到 就把其他所有文件里对应的那一行显示出来
